### ISSUE TITLE
Change handling of JAX arrays, currently cast using `atleast_2d` 

### ISSUE THREAD

**Issue by pc532627** [gchq/coreax#630]
### What's the new feature?

Inside of coreset generation, approximation and other aspects of the codebase, we currently cast arrays `x` that are input using `atleast_2d` ensuring that all computations remain valid. There are tests covering this input format. However, users may be confused as to how data is manipulated (is a one-dimensional length `N` vector converted to a 1 by N array or an N by 1 array, for example). We could consider changing how we handle array casting and use one of the approaches suggested during review of PR #626.

See discussion in https://github.com/gchq/coreax/pull/626 from which the following comment is taken:

We only test for 2D at the moment. I don't like the usage of `atleast_2d` as it is not explicit to the user where this dimension will be added (the leading axis). If the user passes an n-vector (believing this represents n scalar observations) then `atleast_2d` will silently give them the wrong result. This is because `atleast_2d` converts the vector to an array with shape `(1, n)`, rather than `(n, 1)`.

The difficulty is that different users may assume different behaviour for vectors. Some may (correctly in this case) assume an n-vector represents a single n-dimensional observation, whereas others may (incorrectly in this case) assume an n-vector represents n 1-dimensional observations.

Because of this, I think we only have three options:

1. throw an error when passed an array that is not 2D `if jnp.ndim(x) != 2: raise ValueError("Must be 2D")`
2. allow k-dimensional arrays and let JAX handle any resulting incompatible mathematical operations.
3. use beartype and jaxtyping to provide (optional) runtime type checking of the array dimensions.
If we start using jaxtyping everywhere then the expectations on user data become clearer before runtime (at programming time).

Addressing this issue properly across the entire codebase is out of scope for this pull, so I would prefer to keep this as is, or just wrap the `asarrays` in an `atleast_2d` just to preserve the previous behaviour (whichever you would prefer)

### What value does this add?

Clarity to the user on how input data is handled.
